lynx/src/main/java/com/github/pedrovgs/lynx/model/Lynx.java

```diff
@@ -143,10 +161,17 @@ private boolean shouldAddTrace(String logcatTrace) {
   }
 
   private synchronized boolean traceMatchesFilter(String logcatTrace) {
-    TraceLevel levelFilter = lynxConfig.getFilterTraceLevel();
-    String filter = lynxConfig.getFilter().toLowerCase();
-    String logcatTraceLowercase = logcatTrace.toLowerCase();
-    return logcatTraceLowercase.contains(filter) && containsTraceLevel(logcatTrace, levelFilter);
+    return traceStringMatchesFilter(logcatTrace)
+            && containsTraceLevel(logcatTrace, lynxConfig.getFilterTraceLevel());
+  }
+
+  private boolean traceStringMatchesFilter(String logcatTrace) {
+    String lowerCaseLogcatTrace = logcatTrace.toLowerCase();
+    boolean matchesFilter = lowerCaseLogcatTrace.contains(lowerCaseFilter);
+    if (!matchesFilter && regexpFilter != null) {
```

Comment:
Should we check just the regex if the ``regexpFilter`` value is not null? I guess if you have a valid regex you don't want to see all the tokens with the same value you use in the regex.